smartsim/_core/control/controller.py

```diff
@@ -512,6 +512,14 @@ def _launch_step(
         :raises SmartSimError: if launch fails
         """
         try:
+            if (
+                entity.name in self._jobs.completed
+                or self._jobs.jobs
+                or self._jobs.db_jobs
+            ):
+                raise SSUnsupportedError(
+                    "SmartSim entities cannot have duplicate names."
+                )
             job_id = self._launcher.run(job_step)
```

Comment:
Can we move this out the `try` block, just to keep the area from where we expect errors to be raised as small as possible?
```suggestion
        if (
            entity.name in self._jobs.completed
            or self._jobs.jobs
            or self._jobs.db_jobs
        ):
            raise SSUnsupportedError(
                "SmartSim entities cannot have duplicate names."
            )
        try:
            job_id = self._launcher.run(job_step)
```